src/nbc/nbc.js

```diff
@@ -0,0 +1,103 @@
+const puppeteer = require("puppeteer");
+
+const {
+  makeAddArticlesAction,
+  makeUpdateArticleNormalContentAction,
+  makeUpdateArticleVideoContentAction
+} = require("../store/actions");
+const { makeArticlesWithoutContentSelector } = require("../store/selectors");
+const { store, saveStore } = require("../store/index");
+const { NBC } = require("../constant");
+const { complement, partition, or } = require("../utils");
+
+const {
+  isNbcTextArticleLink,
+  isNbcBetterHref,
+  isNbcFeatureNbcOutHref
+} = require("./nbc-utils");
+
+const addArticles = makeAddArticlesAction(NBC);
+const updateArticle = makeUpdateArticleNormalContentAction(NBC);
+const articlesWithoutContent = makeArticlesWithoutContentSelector(NBC);
+
+const NBC_URL = "https://www.nbcnews.com";
+
+const articleContentUpdate = async page => {
```

Comment:
Sort of similar logic to the CNN article collection. The logic is annoyingly intricate...